skimage/measure/profile.py

```diff
@@ -86,14 +90,31 @@ def profile_line(image, src, dst, linewidth=1,
            [1.        , 1.        , 0.        ],
            [1.41421356, 1.41421356, 0.        ]])
     """
+    image_is_bool = image.dtype == bool
+
+    if order is None:
+        if image_is_bool:
+            order = 0
+        else:
+            order = 1
+
+    if image_is_bool and order != 0:
+        warn("Input image dtype is bool. Interpolation is not defined "
+             "with bool data type. Please set order to 0 or explicitely "
+             "cast input image to an other data type.")
+
+    prefilter = order > 1
+
     perp_lines = _line_profile_coordinates(src, dst, linewidth=linewidth)
     if image.ndim == 3:
         pixels = [ndi.map_coordinates(image[..., i], perp_lines,
-                                      order=order, mode=mode, cval=cval)
-                  for i in range(image.shape[2])]
+                                      prefilter=prefilter,
```

Comment:
Again, I don't think it's necessary to define `prefilter`, just write `prefilter=(order > 1),` in the function call.